File: ReportingModule.API.developer7/NsbWeb.ReportingModule/ViewModels/ReportingPeriodViewModel.cs
using System;
using Newtonsoft.Json;
using ReportingModule.ValueObjects;

namespace NsbWeb.ReportingModule.ViewModels
{
    public class ReportingPeriodViewModel
    {
        [JsonIgnore]
        public ReportingPeriod ReportingPeriodData { get; set; }
        public OrganizationReference OrganizationReference { get; set; }
        public int Year => ReportingPeriodData.Year;
        public ReportingFrequency ReportingFrequency => ReportingPeriodData.ReportingFrequency;
        public ReportingTerm ReportingTerm => ReportingPeriodData.ReportingTerm;
        public DateTime StartDate => ReportingPeriodData.StartDate;
        public DateTime EndDate => ReportingPeriodData.EndDate;
        public bool IsActive { get; set; }
    }
}

File: ReportingModule.UI.developer7/src/models/ReportingPeriodViewModel.ts
import { OrganizationReference } from "models/OrganizationReference";
import { ReportingFrequency } from "models/ReportingFrequency";
import { ReportingTerm } from "models/ReportingTerm";

export class ReportingPeriodViewModel {
    constructor(public organizationReference: OrganizationReference, 
        public year: number,
        public reportingFrequency: ReportingFrequency, 
        public reportingTerm: ReportingTerm, 
        public startDate: Date, 
        public endDate: Date, 
        public isActive: boolean) { }
}
